### ISSUE TITLE
Potential false positives on BooleanPropertyNaming with constants

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#4920]
## Context
BooleanPropertyNaming considers naming of properties, while technically these `const val`s are properties, they also have a different naming convention and I have a feeling they rarely start with `IS_` prefix.

## Expected Behavior of the rule
I propose to exclude `const` from this check fully.

```kotlin
object WidgetPreferences { // Detekt: or a companion object

	/**
	 * User selection of whether to show the part of day or not.
	 *
	 * *type*: [Boolean]
	 * *description*: true=show
	 * *default*: [DEFAULT_SHOW_PART_OF_DAY]
	 */
	const val PREF_SHOW_PART_OF_DAY = "showPartOfDay"
	const val DEFAULT_SHOW_PART_OF_DAY = true // Detekt: BooleanPropertyNaming shouldn't be reported
}
```

**Comment by cortinico** [comment]
Yup that's legit from my point of view 👍

**Comment by gouri-panda** [comment]
May I try this?

**Comment by cortinico** [comment]
Sure @gouri-panda, I've assigned this to you :+1:

**Comment by amitdash291** [comment]
As there is no recent activity on this, I am taking this up.

**Comment by TWiStErRob** [comment]
Thank you @amitdash291 !
